homeassistant/components/lutron/light.py

```diff
@@ -25,14 +31,38 @@ async def async_setup_entry(
     Adds dimmers from the Main Repeater associated with the config_entry as
     light entities.
     """
+    ent_reg = er.async_get(hass)
     entry_data: LutronData = hass.data[DOMAIN][config_entry.entry_id]
-    async_add_entities(
-        [
-            LutronLight(area_name, device, entry_data.client)
-            for area_name, device in entry_data.lights
-        ],
-        True,
-    )
+
+    for area_name, device in entry_data.lights:
+        if device.type == "CEILING_FAN_TYPE":
+            # If this is a fan check to see if this entity already exists. If not, do not create a new one.
+            entity_id = f"light.{slugify(f'{area_name} {device.name}')}"
+            if ent_reg.async_is_registered(entity_id):
+                entity_entry = ent_reg.async_get(entity_id)
+                assert entity_entry
+                if entity_entry.disabled:
+                    # If the entity exists and is disabled then we want to remove the entity so that the user is using the new fan entity instead.
+                    ent_reg.async_remove(entity_id)
+                else:
+                    async_add_entities(
```

Comment:
Please make a single call to add all the entities. Collect the entities in a list first.